lib/liquid/template.rb

```diff
@@ -64,11 +62,11 @@ class << self
       attr_writer :taint_mode
 
       def file_system
-        @@file_system
+        Thread.current[:liquid_file_system] ||= BlankFileSystem.new
```

Comment:
Ok, seems fair. Maybe it would be best to deprecate this class-level accessor then and use instead: `context.register.fetch(: file_system) { BlankFileSystem.new }` so that no crash between threads could happen. This PR I guess can be closed.